[DX] PR Change the way that compiling looks like

CosmeValera opened the pull request:
I added the configuration changes to improve the way that the compiling looks for webpack and rspack.

## 📷 RSPACK
- Before:
![image](https://github.com/jherr/create-mf-app/assets/80126839/b6455584-d76a-46fb-afc6-4a8a58db0a18)

- After:
![image](https://github.com/jherr/create-mf-app/assets/80126839/820c715c-c1a7-49dd-ac87-13e13cb2d5a1)

Feel free to try the changes, and let me know if you need anything else, or if there is something that you don't like 💓💓

<small>(This PR is linked to https://github.com/jherr/create-mf-app/issues/68)</small>

## Additional thing
I also added this line inside `module.exports.devServer`:
```js
watchFiles: [path.resolve(__dirname, 'src')],
```
![image](https://github.com/jherr/create-mf-app/assets/80126839/e64b8608-0992-47b0-a0b6-ae93a50fae82)`

It solves a bug/issue regarding remote mfes not having HMR.
I found I problem where using your `create-mf-app` library for creating a host and a remote app. Then changes in the remote or the host will have HMR (Hot Module Replacemente) in the host, but not in the remote. Like in this image:
![image](https://github.com/jherr/create-mf-app/assets/80126839/eaf9d18e-7918-4e7e-8f1a-b422664e839a)
You can use the code of this mini sample to try to check it if you want: https://github.com/CosmeValera/ReactProjects/tree/master/micro-fes/micro-fe-basic-sample-of-host-and-remote-linked
or if you want we can talk about it somewhere else, in another PR/Issue, but yes, basically that line fixes this problem of the remote not having HMR. I found this code snippet researching for having hmr with webpack, however it seems that it makes the program slightly slower, not that I notice it at all, but maybe it could cause problem in some scenario.

CosmeValera commented:
## ⚠️ Not related ISSUES ⚠️
I also found 2 things.
- react-esm does not have `rspack` config file. So for what I understand whenever you pick react-esm and rspack it picks webpack anyway. I don't know if this behaviour is intended in the sense that rspack is not compatible with react-esm, but if it was an oversight, I feel that the rspack file for react-sm must be very similar as the one for react.
- When executing any script in a `rspack` repo, an error regarding not understanding 'NODE_ENV' would arise in my terminal, maybe i did something wrong, I don't know anything about rspack :_(, I just did `npm i && npm start`.

I'm happy about you giving me more feedback, I'm open to keep helping the repo with what I can 💓💓

jherr commented:
Let's see if we can pull in @ScriptedAlchemy to give us some additional insights into rspack behaviors.

CosmeValera commented:
In my case what I did to try the rspack versions was to remove the NODE_ENV part in the scripts in the package.json.

In this case I consider that the goal of the PR is fully addressed with the code given. What I added in the comment starting with ⚠️ Not related ISSUES ⚠️ was matters for other issues/PRs. You can already test out the webpack and rspack apps 😊👍

Chemaclass commented on templates/application/lit-html/base/rspack.config.js:
```diff
@@ -1,6 +1,38 @@
 const isDev = process.env.NODE_ENV === 'development'
+const path = require('path');
+
+const printCompilationMessage = (status, port) => {
+  let messageColor, messageType, browserMessage
+
+  switch (status) {
+    case 'success':
+      messageColor = '\x1b[32m'
+      messageType = 'Compiled successfully!'
+      browserMessage = 'You can now view'
+      break
+    case 'failure':
+      messageColor = '\x1b[31m'
+      messageType = 'Compilation Failed!'
+      browserMessage = 'You can\'t now view'
+      break
+    case 'compiling':
+      messageColor = '\x1b[94m'
+      messageType = 'Compiling...'
+      browserMessage = 'Compiling the'
+      break
+  }
```
Can it be possible to extract this function into another higher level to be able to reuse it in the other 15 places? 
![Screenshot 2024-07-01 at 08 52 10](https://github.com/jherr/create-mf-app/assets/5256287/e9a452a7-73cd-4387-a609-410aba51dcc8)

CosmeValera replied:
Thanks for the idea!
I think this covers it 🥰🥰 https://github.com/jherr/create-mf-app/pull/69#issuecomment-2199946630

CosmeValera commented:
> Let's see if we can pull in @ScriptedAlchemy to give us some additional insights into rspack behaviors.

It was my fault for being messy and talking about rspack here. But I would prefer to touch the rspack issues in another place, and maintain this PR just for the compiling idea. I created this issue for the rspack problem that I mentioned: https://github.com/jherr/create-mf-app/issues/70

CosmeValera commented:
## 🧠 What changed:
In these last commits I extracted the method `printCompilationMessage` to another file which I have added in `templates/compiler/compilation.config.ts`. 
And in `src/index.ts` I added the code to have a dynamic import of the `compilation.config.js` file:
```ts
const copyCompilationConfig = (projectName: string) => {
  const sourcePath = path.join(__dirname, '../templates/compiler/compilation.config.js')
  const destinationPath = path.join(projectName, 'compilation.config.js')

  fs.copyFileSync(sourcePath, destinationPath)
}
```

## 📷 Folder structure

### 1. `create-mf-app` project
![image](https://github.com/jherr/create-mf-app/assets/80126839/d48e0872-851b-4cfe-9926-1a3cc6c6d57c)
`templates/compiler/compilation.config.ts`

### 2. `host` project
![image](https://github.com/jherr/create-mf-app/assets/80126839/d5ec9541-0f2a-4081-aaf2-0fb2faade8c5)
`compilation.config.js`

Chemaclass commented:
ping
